test/Object/ObjectRest_Simple.js

```diff
@@ -0,0 +1,287 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// Object Rest unit tests
+
+if (this.WScript && this.WScript.LoadScriptFile) { // Check for running in ch
+    this.WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+}
+
+var tests = [
+    {
+        name: "let assignment with simple Object",
+        body: function() {
+            let {a, b, ...rest} = {a: 1, b: 2, c: 3, d: 4};
+            assert.areEqual(1, a);
+            assert.areEqual(2, b);
+            assert.areEqual({c: 3, d: 4}, rest);
+        }
+    },
+    {
+        name: "var assignment with simple Object",
+        body: function() {
+            var {a, b, ...rest} = {a: 1, b: 2, c: 3, d: 4};

```

Comment:
test some invalid syntax
```
...(rest)
...rest++ (postfix operators)
...rest+1
let {... ...rest} = {}
let {...rest} = undefined;
let {...} = {}
```

default args for rest?
foo({a =1, ...rest = {} }) {}

Reply:
I originally tried to test for syntax errors, but the examples I found in the tests did not work for syntax errors. Do you know of an example I could follow to write syntax tests?

---
In reply to: [205211644](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r205211644) [](ancestors = 205211644)

Reply:
something like 
```assert.throws```
look at test\es6\destructuring_obj.js for few example

---
In reply to: [205222323](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r205222323) [](ancestors = 205222323,205211644)